src/bokeh/plotting/_plot.py

```diff
@@ -78,16 +79,19 @@
 #-----------------------------------------------------------------------------
 
 def get_range(range_input: Range | tuple[float, float] | Sequence[str] | pd.Series[Any] | GroupBy | None) -> Range:
-    import pandas as pd
-    from pandas.core.groupby import GroupBy
+    if uses_pandas(range_input):
+        import pandas as pd
+        from pandas.core.groupby import GroupBy
+    else:
+        pd = GroupBy = None
 
     if range_input is None:
         return DataRange1d()
-    if isinstance(range_input, GroupBy):
+    if GroupBy and isinstance(range_input, GroupBy):
```

Comment:
```suggestion
    if pd and isinstance(range_input, GroupBy):
```

I think this is better.

Reply:
Then `GroupBy` can be `None` (from typing perspective) after `and`.